src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1639,19 +1637,17 @@ private TokenCacheItem findFamilyItemForUser(final UserIdentifierType userIdenti
             return null;
         }
 
-        Iterator<TokenCacheItem> allItems = mTokenCacheStore.getAll();
-        if (allItems == null || !allItems.hasNext()) {
+        Iterable<TokenCacheItem> allItems = mTokenCacheStore.getAll();
+        if (allItems == null || !allItems.iterator().hasNext()) {
```

Comment:
The original reason I make it Iterator is mainly because DefaultTokenCache not only implement the ITokenCacheStore but also the ITokenStoreQuery, ITokenStoreQuery has the getAll, and it returns the Iterator. 

I agree with you guys, the code needs to be improved a lot. The interface for token cache should be able to do the query. There may be some other clients out there providing their customized cache, and they need to implement the interface, I just want to reduce the changes that could happen to them as little as possible.